// storybook-vscode-component/dist/OpenButton.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _react = _interopRequireDefault(require("react"));

var _api = require("@storybook/api");

var _utils = require("./utils");

var _constants = require("./constants");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

var OpenButton = function OpenButton() {
  var results = (0, _api.useParameter)('storySource', {
    storyPath: '',
    componentPath: '',
    workingDir: '',
    repository: '',
    branch: ''
  });
  var storyPath = results.storyPath,
      componentPath = results.componentPath;
  if (!results || !storyPath && !componentPath) return null;
  var url = (0, _utils.generateUrl)(results);
  if (!url) return null;
  return /*#__PURE__*/_react["default"].createElement("a", {
    href: url,
    target: "_blank",
    rel: "noreferrer"
  }, /*#__PURE__*/_react["default"].createElement("img", {
    style: {
      width: '15px'
    },
    alt: "Open source",
    src: process.env.NODE_ENV === 'development' ? _constants.vscodeIcon : _constants.githubIcon
  }));
};

var _default = OpenButton;
exports["default"] = _default;
